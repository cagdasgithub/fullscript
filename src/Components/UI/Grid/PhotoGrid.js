import Photo from "../../Photo/Photo";
import Grid from "@material-ui/core/Grid";
import React from "react";


const PhotoGrid = props => {
  return props.data ? (
    <Grid container spacing={24} style={{ padding: 24 }}>
      {props.data.photos.map(photo => (
        <Grid item xs={12} sm={6} lg={4} xl={3} key={photo.id}>
          <Photo key={photo.id} photo={photo} />
        </Grid>
      ))}
    </Grid>
  ) : null;
};

export default PhotoGrid;
