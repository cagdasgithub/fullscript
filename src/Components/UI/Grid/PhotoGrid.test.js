import React from "react";
import PhotoGrid from "./PhotoGrid";
import renderer from "react-test-renderer";
import { render, queryAllByAltText, cleanup } from "react-testing-library";
afterEach(cleanup);
describe("Tests for Search controls", () => {
  it("renders correctly", () => {
    const grid = renderer.create(<PhotoGrid />).toJSON();
    expect(grid).toMatchSnapshot();
  });

  it("shows valid data", () => {
    const data = {
      photos: [
        {
          id: "1",
          urls: { regular: "small image 1" },
          links: { download: "download" },
          description: "test",
          user: { name: "Cagdas", location: "Ottawa" }
        },
        {
          id: "2",
          urls: { regular: "small image 2" },
          links: { download: "download" },
          description: "test",
          user: { name: "Cagdas", location: "Ottawa" }
        }
      ]
    };
    const { queryAllByAltText } = render(<PhotoGrid data={data} />);
    const download = queryAllByAltText(/complex/i);
    expect(download.length).toEqual(2);
  });
});
