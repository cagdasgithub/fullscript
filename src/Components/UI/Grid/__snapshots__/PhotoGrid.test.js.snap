// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Search controls renders correctly 1`] = `null`;
